app/src/main/java/org/mozilla/tv/firefox/channels/ChannelRepo.kt

```diff
@@ -89,10 +99,10 @@ class ChannelRepo(
             else -> throw NotImplementedError("other types shouldn't be able remove tiles")
         }
 
-        return _sharedPreferences.getStringSet(sharedPrefKey, Collections.emptySet())!!.toList()
+        return _sharedPreferences.getStringSet(sharedPrefKey, Collections.emptySet())!!
     }
 
-    private fun saveBlackList(source: TileSource, blackList: List<String>) {
+    private fun saveBlackList(source: TileSource, blackList: Set<String>) {
```

Comment:
This commit consists of mostly refactor work. So the root cause of the tile deletion not persisting issue is the usage of `List` instead of `Set`?

Reply:
Most of the changes are actually functional, not code cleanliness, but I didn't have time to format the commits in a more understandable way.  Prior to this commit, bundled tiles were not hooked up to any blacklist, so pinned tile removals were not persisted.  This led to a lot of weird behavior.

The easiest way to hook them up was to use the existing `filterNotBlacklisted` method, but doing so meant moving some code around.  It's nice that some of it wound up in better places, but the goal was to make minimal changes to get the code to work.

The `List` -> `Set` change was my first theory on what was causing problems.  I still suspect there are some edge cases for which it's important, but I was asked to get this fix out by noon, so I didn't stop to verify which change fixed each problem.